### PR TITLE
[feat]: [CDS-42886]:Gar region fetch

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Describe your changes
Added Regions api for google artifact registry
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37419)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37419)

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gar/regionGar.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Data
+@Builder
+@NoArgsConstructor
+public class regionGar {
```
Comment (puthrayaharness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gar/regionGar.java:12]: Please capitalize class name `RegionGar`

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gar/GARArtifactResource.java [hunk 2]
```diff
@@ -60,4 +62,49 @@ public class GARArtifactResource {
         connectorRef, region, repositoryName, project, pkg, version, versionRegex, orgIdentifier, projectIdentifier);
     return ResponseDTO.newResponse(buildDetails);
   }
+  @GET
+  @Path("getRegions")
+  @ApiOperation(value = "Gets google artifact registry regions", nickname = "getRegionsForGoogleArtifactRegistry")
+  public ResponseDTO<List<regionGar>> getRegions() {
+    List<regionGar> regions = new ArrayList<>();
```
Comment (puthrayaharness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gar/GARArtifactResource.java:69]: Please create a static final list of strings and you can initialize it using that.
Something along this line. 

`Lists.newArrayList("asia", "europe"...).stream().map(region -> new Region(region, region)).collect...`

Another way to do this is to have this as part of the config file where we have the regions listed in the config. That way we do not have to make a code change to add a new region we can just update the config file.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gar/GARArtifactResource.java [hunk 3]
```diff
@@ -60,4 +63,22 @@ public class GARArtifactResource {
         connectorRef, region, repositoryName, project, pkg, version, versionRegex, orgIdentifier, projectIdentifier);
     return ResponseDTO.newResponse(buildDetails);
   }
+  @GET
+  @Path("getRegions")
+  @ApiOperation(value = "Gets google artifact registry regions", nickname = "getRegionsForGoogleArtifactRegistry")
+  public ResponseDTO<List<RegionGar>> getRegions() {
+    List<RegionGar> regions =
```
Comment (puthrayaharness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gar/GARArtifactResource.java:70]: Please make this static final like the way I had mentioned before. 
`private static final List<..> GAR_REGIONS = Arrays.asList(...)`

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37419)* status: :shipit: complete! ✓

**Pull request merged**